code/game/machinery/doors/firedoor.dm

```diff
@@ -30,6 +30,17 @@
 	..()
 	CalculateAffectingAreas()
 
+/obj/machinery/door/firedoor/examine(mob/user)
+	..()
+	if(!density)
+		to_chat(user, "<span class='notice'>It is open, but could be <b>pried</b> closed.</span>")
+	else if(density && !welded)

```

Comment:
All of these check for density despite being under the `else` for `if(!density)`; make them not check for density, as it is already checked.